inst/OMOP-Data-Quality.Rmd

```diff
@@ -0,0 +1,969 @@
+  }
+
+```
+
+
+```{r temperature}
+
+temperature = measurements%>% filter(measurement_concept_id== get_source_concept_id('temperature',question_mapping))
+
+abnormal_temp = check_measure_bounds('temperature',measure_bounds,temperature)
+
+check = glue('Temperature is within 32-115 °F level')
+
+if (nrow(abnormal_temp)>0) {
+  fail_count = nrow(abnormal_temp)
+  fail_percentage = format(round(fail_count/nrow(temperature)*100,4), scientific = FALSE)
+  if (as.numeric(fail_percentage) > as.numeric(tolerance)) {
+    data_qual_measure = add_new_check(data_qual_measure,nrow(temperature),check,'Fail',fail_count,fail_percentage)
+  }else{
+    data_qual_measure = add_new_check(data_qual_measure,nrow(temperature),check,'Pass',fail_count,fail_percentage)
+  }
+  } else {
+    data_qual_measure = add_new_check(data_qual_measure,nrow(temperature),check,'Pass','0','0')
+  }
+
+```
+
+```{r repository rate}
+
+respiratory_rate = measurements%>% filter(measurement_concept_id== get_source_concept_id('respiratory_rate',question_mapping))
+
+abnormal_rr = check_measure_bounds('respiratory_rate',measure_bounds,respiratory_rate)
+
+check = glue('Repository rate is within 0-100 bpm level')
+
+if (nrow(abnormal_rr)>0) {
+  fail_count = nrow(abnormal_rr)
+  fail_percentage = format(round(fail_count/nrow(respiratory_rate)*100,4), scientific = FALSE)
+  if (as.numeric(fail_percentage) > as.numeric(tolerance)) {
+    data_qual_measure = add_new_check(data_qual_measure,nrow(respiratory_rate),check,'Fail',fail_count,fail_percentage)
+  }else{
+    data_qual_measure = add_new_check(data_qual_measure,nrow(respiratory_rate),check,'Pass',fail_count,fail_percentage)
+  }
+  } else {
+    data_qual_measure = add_new_check(data_qual_measure,nrow(respiratory_rate),check,'Pass','0','0')
+  }
+
+```
+
+```{r diastolic blood pressure}
+
+diastolic_blood_pressure = measurements %>%filter(measurement_concept_id== get_source_concept_id('diastolic_blood_pressure',question_mapping))
+
+abnormal_dbp = check_measure_bounds('diastolic_blood_pressure',measure_bounds,diastolic_blood_pressure)
+
+check = glue('Diastolic blood pressure rate is within 0-200 mmHg level')
+
+if (nrow(abnormal_dbp)>0) {
+  fail_count = nrow(abnormal_dbp)
+  fail_percentage = format(round(fail_count/nrow(diastolic_blood_pressure)*100,4), scientific = FALSE)
+  if (as.numeric(fail_percentage) > as.numeric(tolerance)) {
+    data_qual_measure = add_new_check(data_qual_measure,nrow(diastolic_blood_pressure),check,'Fail',fail_count,fail_percentage)
+  }else{
+    data_qual_measure = add_new_check(data_qual_measure,nrow(diastolic_blood_pressure),check,'Pass',fail_count,fail_percentage)
+  }
+  } else {
+    data_qual_measure = add_new_check(data_qual_measure,nrow(diastolic_blood_pressure),check,'Pass','0','0')
+  }
+
+```
+
+```{r systolic blood pressure}
+
+systolic_blood_pressure = measurements%>% filter(measurement_concept_id== get_source_concept_id('systolic_blood_pressure',question_mapping))
```

Comment:
You can't keep adding chunks of code for each physiology variable. 
We may potentially end up with 10s to hundreds of these to check. :)
The whole measurements check table needs to be generated based on the rows in the csv.  So each time a new row is added to the csv, there's a new line in the table.